src/device.rs

```diff
@@ -55,6 +85,15 @@ impl Default for Device {
     }
 }
 
+impl std::fmt::Debug for Device {
```

Comment:
Can we not just use the display impl here?

Reply:
I think this is more a rust convention thing. I read somewhere that a recommended practice for library crates is to always provide `Debug` impl, which is why I added the `#![deny(missing_debug_implementations)]`. And `Display` is considered rather optional, but it doesn't hurt to provide both

Reply:
My brain's rusted lol. Misread your comment. Yeah, we could, copilot just generated the same code and I was too lazy to change that